api/src/test/java/org/pmiops/workbench/billing/BillingProjectBufferServiceTest.java

```diff
@@ -479,13 +489,65 @@ public void assignBillingProject_locking() throws InterruptedException, Executio
   @Test
   public void cleanBillingBuffer_creating() {
     DbBillingProjectBufferEntry entry = new DbBillingProjectBufferEntry();
+    entry.setFireCloudProjectName("foo");
+    entry.setStatusEnum(CREATING, this::getCurrentTimestamp);
+    entry.setCreationTime(getCurrentTimestamp());
+    billingProjectBufferEntryDao.save(entry);
+
+    doReturn(new BillingProjectStatus().creationStatus(CreationStatusEnum.CREATING))
+        .when(fireCloudService)
+        .getBillingProjectStatus(entry.getFireCloudProjectName());
+
+    CLOCK.setInstant(AFTER_CREATING_TIMEOUT_ELAPSED);
+    billingProjectBufferService.syncBillingProjectStatus();
+    billingProjectBufferService.cleanBillingBuffer();
```

Comment:
If we expect, in reality, that some more time elapses between the sync and the clean, then it wouldn't hurt to advance the clock some in between these two events. It could potentially catch a bug if somehow things were better when the two events were nearly simultaneous. (Though I imagine it's more common for race conditions to happen when events like this are happening near each other.)

Bonus question: is it safe for more than one of these crons to be operating on an entry at the same time? I haven't looked at how they're scheduled to see if that's possible.

Reply:
I'd rather keep these tests lean and on-topic. These crons are intended to be safe to operate in parallel (and yes, it is possible). In my view this is sufficient to cover the regression as written, though clearly you could go much further here and add coverage of all permutations of clean/sync ordering with varying time between each.